options.go

```diff
@@ -42,6 +42,7 @@ type dialerConfig struct {
 	refreshTimeout         time.Duration
 	useIAMAuthN            bool
 	logger                 debug.Logger
+	lazyRefresh            bool
```

Comment:
do we want this as `lazyRefresh` or `refreshType`? do we see a future where we add another option for refreshes?

Reply:
As discussed elsewhere, we're keeping this field unexported so we can adapt it in the future if needed.